datasets/pec/pec.py

```diff
@@ -0,0 +1,220 @@
+        is_speaker = True
+        sentences = []
+        with open(path, encoding="utf-8") as f:
+            for row in f:
+                if "********************" not in row:
+                    if is_speaker:
+                        speaker = row.strip()
+                        is_speaker = False
+                    else:
+                        sentences.append(row.strip())
+                else:
+                    persona[speaker] = sentences
+                    is_speaker = True
+                    sentences = []
+        return persona
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO: Downloads the data and defines the splits
+        # dl_manager is a datasets.download.DownloadManager that can be used to
+        # download and extract URLs
+        dl_dir = dl_manager.download_and_extract(_URL)
+        data_dir = os.path.join(dl_dir, "hf_pec")
+        print("the downloaded data has been saved to ", data_dir)
+        if self.config.domain in ["happy", "offmychest"]:
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    gen_kwargs={"filepath": os.path.join(data_dir, self.config.domain, "train.txt"), "split": "train"},
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    gen_kwargs={"filepath": os.path.join(data_dir, self.config.domain, "test.txt"), "split": "test"},
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.VALIDATION,
+                    gen_kwargs={"filepath": os.path.join(data_dir, self.config.domain, "valid.txt"), "split": "dev"},
+                ),
+            ]
+        else:
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    gen_kwargs={"filepath": os.path.join(data_dir, "train.txt"), "split": "train"},
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    gen_kwargs={"filepath": os.path.join(data_dir, "test.txt"), "split": "test"},
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.VALIDATION,
+                    gen_kwargs={"filepath": os.path.join(data_dir, "valid.txt"), "split": "dev"},
+                ),
+            ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        # TODO: Yields (key, example) tuples from the dataset
+        context_speakers = []
+        context = []
+        response_speaker = ""
+        response = ""
+        example_id = 0
+        data_dir = os.path.dirname(filepath)
+
+        if self.config.domain == "all" and not os.path.exists(os.path.join(data_dir, "train.txt")):
+            # concatenate two domain files
+            self._concatenate_files(

```

Comment:
Could you iterate through the files one by one instead of concatenating them ?
The dataset builder is not supposed to write files